### PR TITLE
Add backslash support to registry resource

### PR DESCRIPTION
**Pull request by sbrito85**
Puppet bug here: https://tickets.puppetlabs.com/browse/MODULES-2957

Picked up the work being done on other pull requests.

### CODE REVIEW

**Comment by sbrito85** [comment]
Looks like I need to clean up some tests :)

**Comment by jpogran** [comment]
@sbrito85 thanks for the PR! I've added this to triage and it will be reviewed this week. While that happens, could you review the [CONTRIBUTING.md](https://github.com/puppetlabs/puppetlabs-registry/blob/master/CONTRIBUTING.md) file and update your commits? Specifically referring to commit message section.

File: lib/puppet/type/registry_key.rb [hunk 1]
```diff
@@ -103,7 +103,7 @@ def eval_generate
 
     # get the "should" names of registry values associated with this key
     should_values = catalog.relationship_graph.direct_dependents_of(self).select {|dep| dep.type == :registry_value }.map do |reg|
-      PuppetX::Puppetlabs::Registry::RegistryValuePath.new(reg.parameter(:path).value).valuename
+      PuppetX::Puppetlabs::Registry::RegistryValuePath.new(reg.parameter(:path).value, reg.parameter(:value_name).value).valuename
```
Comment (jpogran) [inline lib/puppet/type/registry_key.rb]: Does this mean it's a breaking change? As in, do users using the old syntax have to use this new syntax?

Comment (sbrito85) [inline lib/puppet/type/registry_key.rb]: Old syntax still works from all tests that I've done. This should not be a breaking change.

**Comment by glennsarti** [comment]
Hi @sbrito85.  Thanks for all your work on this so far.  I have some concerns about it;

* Needs documentation update to explain how to use a value with a name with '\`
* Needs spec tests for valuenames with a `\`
* I briefly started adding some tests and noticed there are some issues with the auto-require logic as it doesn't check for resources with `value_name`

**Comment by sbrito85** [comment]
I plan on having some cycles this week to look at these concerns. Sorry for the delay!

1) Will add documentation, should have that in today or tomorrow
2) For this part, I may need some assistance. 
3) Where should I be checking for this? I presume it should go in lib/puppet/type/registry_value.rb like the path value?

**Comment by glennsarti** [comment]
@sbrito85 

I started work on squashing your commits, and then adding tests.

Commits are here;
https://github.com/glennsarti/puppetlabs-registry/commits/ticket/master/MODULES-2957-backslash

This is what tripped me up with autorequire as those tests were failing.

The autorequire code is here
https://github.com/puppetlabs/puppetlabs-registry/blob/master/lib/puppet/type/registry_value.rb#L124-L137

Note it's using the resource name not the non-backslash value.

**Comment by michaeljmatthews22** [comment]
Any update on this? Really looking forward to this being available for our company.

**Comment by makomatic** [comment]
Any Update? This would reallly be nice...

**Comment by Iristyle** [comment]
This is being superseded in #143 and #142 - as such, closing.

Thanks for the original PR @sbrito85!

**Pull request closed without merging**